hardhat.config.js

```diff
@@ -35,11 +35,20 @@ const optimizerSettings = {
     },
 };
 
+// Need to manually specify outputSelection settings for the hardhat-storage-layout plugin since we're using customized compiler settings.
+// https://www.npmjs.com/package/hardhat-storage-layout?activeTab=code
+const outputSelectionSettings = {
+    "*": {
+        "*": ["storageLayout"],
+    },
+};
+
 const defaultSettings = {
     version: '0.8.23',
     settings: {
         evmVersion: process.env.EVM_VERSION || 'london',
         optimizer: optimizerSettings,
+        outputSelection: process.env.STORAGE_LAYOUT ? {} : outputSelectionSettings,
```

Comment:
don't you want the opposite if then check
```suggestion
        outputSelection: process.env.STORAGE_LAYOUT ? outputSelectionSettings : {},
```

Reply:
Yeah, I noticed that it should be the opposite, that's reason why I deleted my comment on slack but you already replied back, thank you. updated